libsabledb/src/commands/generic_commands.rs

```diff
@@ -170,6 +174,34 @@ impl GenericCommands {
         };
         Ok(Some(md.value_type()))
     }
+
+    async fn expire(
+        client_state: Rc<ClientState>,
+        command: Rc<RedisCommand>,
+        response_buffer: &mut BytesMut,
+    ) -> Result<(), SableError> {
+        let builder = RespBuilderV2::default();
+        // at least 2 arguments
+        check_args_count!(command, 2, response_buffer);
+
+        let db_id = client_state.database_id();
+
+        // EXPIRE key seconds
+        let key = command_arg_at!(command, 1);
```

Comment:
since we have an optional argument in this command, I would take the iterator approach here:

```rust
        check_args_count!(command, 3, response_buffer);
        let builder = RespBuilderV2::default();
        let mut iter = command.args_vec().iter();
        iter.next(); // skip "expire"

       // check the remaining arguments
        match (iter.next(), iter.next(), iter.next()) {
            (Some(key), Some(seconds), Some(arg)) => {
                let arg_lowercase = BytesMutUtils::to_string(arg).to_lowercase();
                match arg_lowercase.as_str() {
                    "nx" => {
                        // ..
                    }
                    "xx" => {
                        // ..
                    }
                    "gt" => {
                        // ..
                    }
                    "lt" => {
                        // ..
                    }
                    _ => {
                        // prepare error message to the user
                        builder.error_string(response_buffer, "...");
                        return Ok(());
                    }
                }
            }
            (Some(key), Some(seconds), None) => {
                // No third argument was provided
            }
            _ => {
                // Anything else is just syntax error (this should be checked against redis output) 
                builder.error_string(response_buffer, ErrorStrings::SYNTAX_ERROR);
                return Ok(());
            }
        }
```